### ISSUE TITLE
Unable to open parameter optimization BB

### ISSUE THREAD

**Issue by teutonicod** [Open-Systems-Pharmacology/PK-Sim#1109]
Dear all,

I am not anymore able to open the results of a parameter identification in PK-Sim. This already occurred me a couple of times, but I not sure which behavior generates this issue. I typically run the optimization, I rename the optimization BB and then I keep working with the model, opening and closing several simulations and optimization and everything works fine. Later on, I think when reopening the PK-Sim project, the optimization does not open anymore and gives me an error. But this does not happe systematically, with all the optimizations. Here below are the details of the error generated; do you have any clue what can generate this behavior?

By the way, the link provided in the error window (http://www.open-systems-pharmacology.org/pk-sim/issues) is not active, I think it should be updated.

Thanks a lot in advance for your suggestion!

Best,

Donato

*Application:*
*PK-Sim® 7.4.0.127*

*Object reference not set to an instance of an object.*

**Stack trace:**
```
at OSPSuite.Presentation.Presenters.ParameterIdentifications.ParameterIdentificationConfidenceIntervalPresenter.confidenceIntervalDTOFrom(ICache`2 confidenceInterval, IdentificationParameter identificationParameter, OptimizedParameterValue optimizedParameter)
   at OSPSuite.Presentation.Presenters.ParameterIdentifications.ParameterIdentificationConfidenceIntervalPresenter.<>c__DisplayClass3_0.<mapFrom>b__2(<>f__AnonymousType0`2 <>h__TransparentIdentifier0)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at OSPSuite.Presentation.Presenters.ParameterIdentifications.ParameterIdentificationConfidenceIntervalPresenter.CalculateConfidenceIntervalFor(ParameterIdentification parameterIdentification, ParameterIdentificationRunResult runResult)
   at OSPSuite.Presentation.Presenters.ParameterIdentifications.ParameterIdentificationCovarianceAnalysisPresenter.UpdateAnalysis()
   at OSPSuite.Presentation.Presenters.ParameterIdentifications.ParameterIdentificationMatrixAnalysisPresenter`1.set_SelectedRunResults(ParameterIdentificationRunResult value)
   at OSPSuite.Presentation.Presenters.ParameterIdentifications.ParameterIdentificationMatrixAnalysisPresenter`1.UpdateAnalysisBasedOn(IAnalysable analysable)
   at OSPSuite.Presentation.Presenters.ParameterIdentifications.ParameterIdentificationMatrixAnalysisPresenter`1.InitializeAnalysis(ISimulationAnalysis simulationAnalysis, IAnalysable analysable)
   at OSPSuite.Presentation.Presenters.EditAnalyzablePresenter`4.createAnalysis(ISimulationAnalysis simulationAnalysis)
   at OSPSuite.Utility.Extensions.EnumerableExtensions.Each[T](IEnumerable`1 list, Action`1 action)
   at OSPSuite.Presentation.Presenters.EditAnalyzablePresenter`4.Edit(TAnalyzable analyzable)
   at OSPSuite.Presentation.Presenters.SubjectPresenter`3.Edit(Object objectToEdit)
   at PKSim.Presentation.Core.SingleStartPresenterTask.StartForSubject[T](T subject)
   at OSPSuite.Presentation.UICommands.EditParameterIdentificationUICommand.PerformExecute()
   at OSPSuite.Presentation.UICommands.ObjectUICommand`1.Execute()
   at OSPSuite.Presentation.UICommands.UICommandExtensions.ExecuteWithinExceptionHandler(IUICommand uiCommand, IEventPublisher eventPublisher, IChangePropagator changePropagator)
```

**Comment by msevestre** [comment]
@teutonicod @Yuri05 I have moved this issue to the PK-Sim repo as it is a bug report related to PK-Sim

**Comment by msevestre** [comment]
>I saved several snapshot in the last weeks, but I cannot load any of them (except just the first one created) since I get again the same error. 

At least with the snapshot, we could try to understand WHAT is missing

> I thought that snapshot could be used as backup, but if one component of the model is corrupt (simulation, PI,...) the whole upload gives an error?

Snapshot are used to recreate model from scratch based on user inputs. IT will recreate brand new Building blocks, simulations etc.. That means that snapshot will always be using the latest and greatest `internal` structure of a simulation. More specifically, if the internal structure changes between two versions, the snapshot will create a newer model that might be completely equivalent to the old one

In short a snapshot recreates a NEW project as if the user were to enter everything by hand again. If the PK-Sim versions differs, you might get different output. So I would NOT use this system as backup

**Comment by Yuri05** [comment]
@teutonicod So what are the error messages with loading from snapshot? Same as here (PI issue)?
Would it be possible to extract only the problematic PI node(s) from project snapshots and attach them here?

**Comment by teutonicod** [comment]
@msevestre Thank you very much for the clarification, indeed the snapshot would do the job on a short term (within the same PK-Sim version if I gig it right) but not on the long term
@Yuri05 This is the error I get loading the snapshot

Information: Loading Project from snapshot file 
Error: Object reference not set to an instance of an object.

For more information or questions, please visit the Open Systems Pharmacology Forum (forum.open-systems-pharmacology.org).

Stack trace:
at OSPSuite.Core.Domain.Services.ParameterIdentifications.IdentificationParameterFactory.<>c.<ensureThatDimensionsAreEqual>b__8_0(IParameter x)
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.Linq.Enumerable.<DistinctIterator>d__64`1.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at OSPSuite.Core.Domain.Services.ParameterIdentifications.IdentificationParameterFactory.ensureThatDimensionsAreEqual(IEnumerable`1 parameters)
   at OSPSuite.Core.Domain.Services.ParameterIdentifications.IdentificationParameterFactory.CreateFor(IEnumerable`1 parameterSelections, ParameterIdentification parameterIdentification)
   at PKSim.Core.Snapshots.Mappers.IdentificationParameterMapper.<MapToModel>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.SnapshotMapperBaseExtensions.<MapTo>d__9`2.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.ParameterIdentificationMapper.<MapToModel>d__8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.SnapshotMapperBaseExtensions.<MapTo>d__9`2.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.ProjectMapper.<MapToModel>d__11.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.SnapshotMapperBase`2.<MapToModel>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Services.SnapshotTask.<LoadModelFromSnapshot>d__14`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Services.SnapshotTask.<LoadProjectFromSnapshot>d__15.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Presentation.Presenters.Snapshots.LoadProjectFromSnapshotPresenter.<LoadModelAsync>d__4.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Presentation.Presenters.Snapshots.LoadFromSnapshotPresenter`1.<PerformLoad>d__19.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at PKSim.Presentation.Presenters.Snapshots.LoadFromSnapshotPresenter`1.<Start>d__14.MoveNext()

**Comment by msevestre** [comment]
Ok I could reproduce the issue
Bug in PKSim, but also slight user nonsense :)

**Comment by msevestre** [comment]
closing in favor of the issue in core

**Comment by msevestre** [comment]
actually will keep it open until we implement it in core

**Comment by msevestre** [comment]
@teutonicod Can you verify that the issue is fixed with v9? Thanks
